File: 1887-reduction-operations-to-make-the-array-elements-equal/1887-reduction-operations-to-make-the-array-elements-equal.cpp
class Solution {
public:
    int reductionOperations(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int cnt = 0, mx = nums.back();
        for(int i=nums.size()-2; i>=0; i--){
            if(nums[i] != mx){
                mx = nums[i];
                cnt += nums.size()-i-1;
            }
        }
        return cnt;
    }
};

File: 1887-reduction-operations-to-make-the-array-elements-equal/1887-reduction-operations-to-make-the-array-elements-equal.py
class Solution:
    def reductionOperations(self, nums: List[int]) -> int:
        nums.sort();
        cnt, up = 0, 0
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1]: up+=1
            cnt += up
        return cnt